packages/shell-chrome/src/backend.js

```diff
@@ -113,10 +115,17 @@ function serializeDataProperty(value) {
     }
 }
 
-function discoverComponents(isThroughMutation = false) {
+let components = []
+
+function discoverComponents() {
     var rootEls = document.querySelectorAll('[x-data]')
 
-    var components = []
+    // Exit early if no components have been added or removed
+    if (components.length === rootEls.length) {
```

Comment:
this won't work if a component is replaced right?

eg. I'm adding and removing a component at the same time?

Potientially we should keep this check but also add a "if the ids in `components` don't match those in `rootEls`"?